Sources/FunctionalTableData/CollectionView/SupplementaryConfig.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  SupplementaryConfig.swift
+//  FunctionalTableData
+//
+//  Created by Jason Kemp on 2021-10-01.
+//  Copyright © 2021 Shopify. All rights reserved.
+
+import UIKit
+
+public struct Supplementary<View>: CollectionSupplementaryItemConfig where View: UICollectionReusableView {
```

Comment:
Originally, I had only `SupplementaryConfig<View, State>` (which is an analog to TableHeaderFooterConfig in the table view world). It was supposed to represent any supplementary view with a corresponding state type, exactly like our cells work. However collection view has supplementary views whose set of functionality goes beyond headers and footers to include badges, accessory views, separators, content backgrounds, global header footers.

In the content background case (which we'll use almost immediately), it's only a view, it can't be configured the way headers or footers can. Hence, `Supplementary<View>` which is solely for declaring a background view. Then I found while implementing separators that for supplementary views associated with items, we would want per item state, hence `IndexableSupplementaryConfig<View, State>`